### PR TITLE
Relative actions

### PR DESCRIPTION
**Pull request by anthonysimeonov**

### CODE REVIEW

File: src/behavior/base.py [hunk 1]
```diff
@@ -330,6 +332,28 @@ def _sample_action_pred(self, nobs):
...
+            action_pred[:, :, :3] += curr_pos[:, None, :]
+
+            # Each action in the chunk will be relative to the current EE pose
+            curr_ori_quat = C.rotation_6d_to_quaternion_xyzw(curr_ori_6d)
+
+            # Calculate the relative rot action
+            action_quat = C.rotation_6d_to_quaternion_xyzw(action_pred[:, :, 3:9])
+
+            # Apply the relative quat on top of the current quat to get the absolute quat
+            action_quat = C.quaternion_multiply(curr_ori_quat[:, None], action_quat)
```
Comment (anthonysimeonov) [inline src/behavior/base.py:352]: We need to double check if right multiplying by the rotation here is correct. It might slightly depend on what we want to do with passing the current EE pose to the policy.

File: src/dataset/dataset.py [hunk 2]
```diff
@@ -267,7 +284,41 @@ def __getitem__(self, idx):
...
+            # Want a quaternion such that if it's applied to the current EE pose, it will result in the action (absolute pose)
+            # This is the same as the relative rotation between the current EE pose and the action
+            # curr_quat * rel_quat = action_quat_xyzw -> rel_quat = curr_quat^-1 * action_quat_xyzw
+            action_quat_xyzw = C.quaternion_multiply(
+                C.quaternion_invert(curr_ee_quat_xyzw), action_quat_xyzw
+            )
+
+            nsample["action"][:, 3:9] = C.quaternion_to_rotation_6d(action_quat_xyzw)
+
+            # Normalize the relative actions
```
Comment (anthonysimeonov) [inline src/dataset/dataset.py:318]: Is this normalizing twice?

File: src/behavior/base.py [hunk 3]
```diff
@@ -330,6 +332,28 @@ def _sample_action_pred(self, nobs):
         # (B, pred_horizon, action_dim)
         action_pred = self.normalizer(naction, "action", forward=False)
 
+        # These actions may be `pos`, `delta`, or `relative`, if relative, we make them absolute
+        if self.action_type == "relative":
+            # Need the current EE position in the robot frame unnormalized
+            curr_pose = self.normalizer(nobs[:, :16], "robot_state", forward=False)[
```
Comment (anthonysimeonov) [inline src/behavior/base.py:338]: Needed to change to this locally to make this work:

```
            curr_pose = self.normalizer(nobs[..., :16], "robot_state", forward=False).reshape(-1, 16)[
                :, :9
            ]
```

not sure if there's a better way (mainly to account for the potential extra dimension when we get a prediction from the transformer policy)

File: src/eval/evaluate_model.py [hunk 4]
```diff
@@ -329,10 +332,11 @@ def convert_state_dict(state_dict):
 
                 cfg = OmegaConf.create(run.config)
 
-                # Check that we didn't set the wrong action type above
-                assert cfg.control.control_mode == args.action_type, (
-                    f"Control mode in the config: {cfg.control.control_mode} "
-                    f"does not match the action type: {args.action_type}"
+                # Check that we didn't set the wrong action type and pose representation
+                assert cfg.control.control_mode == args.action_type
+                assert (
```
Comment (anthonysimeonov) [inline src/eval/evaluate_model.py:337]: May potentially want to wrap this in an `if` to keep backward compatibility with previous configs we may still want to evaluate

File: src/gym/__init__.py [hunk 5]
```diff
@@ -110,6 +112,8 @@ def get_rl_env(
         # Filter out keys with `image` in them
         obs_keys = [key for key in obs_keys if "image" not in key]
 
+    action_type = "pos" if action_type == "relative" else action_type
```
Comment (anthonysimeonov) [inline src/gym/__init__.py:115]: Let's print out a warning that we're doing this here, since it could be confusing to a user who thinks that changing to "relative" actually does something to the underlying env
